src/api_types.jl

```diff
@@ -253,8 +253,8 @@ end
 # Dataspace constants
 @defconstants H5S begin
     # atomic data types
-    ALL::hid_t       = 0
-    UNLIMITED::hid_t = typemax(hsize_t)
+    ALL::hsize_t       = 0
+    UNLIMITED::hsize_t = typemax(hsize_t)
```

Comment:
nice catch

Reply:
It's actually a bit ambiguous if you read `H5Spublic.h`:
```
#define H5S_ALL         (hid_t)0
#define H5S_UNLIMITED   HSIZE_UNDEF
```
As far as I can tell, though, `hsize_t` makes as much or more sense, though.

Reply:
I guess you're right, seems like it was changed from 1.18  https://github.com/HDFGroup/hdf5/blob/1.8/master/src/H5Spublic.h#L24 
https://github.com/HDFGroup/hdf5/blob/1.10/master/src/H5Spublic.h#L25

Reply:
We don't have to — what I meant by ambiguous was that I couldn't quickly/easily tell (from headers nor documentation) whether `H5S_ALL` and `H5S_UNLIMITED` are supposed to be essentially two possible values for the same type of thing or if the former actually has multiple meanings. I'm fine with just going with the literal translation of the header, though.